### PR TITLE
Bug 1145758 - remove old endedTimer workaround

### PR DESCRIPTION
**Pull request by davidflanagan**

### CODE REVIEW

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=e1afa571741ab6af24e52c83f5531c2e76ed718d

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=f72a4d318475620445e11f7aa36e57b3a778f237

File: apps/music/js/ui/views/player_view.js [hunk 1]
```diff
@@ -590,8 +581,19 @@ var PlayerView = {
         this.setAudioSrc(this.dataSource);
       }.bind(this));
     } else {
-      // If we reach here, the player is paused so resume it
-      this.audio.play();
+      // If we reach here, the player is paused so resume it.
+      // But we're very close to the end of the song, then just
```
Comment (jimporter) [inline apps/music/js/ui/views/player_view.js]: I think you're missing a word here. "But *if* we're..."?

File: apps/music/js/ui/views/player_view.js [hunk 2]
```diff
@@ -590,8 +581,19 @@ var PlayerView = {
...
+      // skip to the next song rather than finishing this one.
+      // (This works around a bug where if we're within a fraction of
+      // a second of the end of a .m4a file, it takes a long time to
+      // get an ended event and move to the next song)
+      if (this.audio.duration > 20 &&
```
Comment (jimporter) [inline apps/music/js/ui/views/player_view.js]: Why the 20 here? Does it actually matter how long the track is?

File: apps/music/js/ui/views/player_view.js [hunk 3]
```diff
@@ -590,8 +581,19 @@ var PlayerView = {
...
+          this.audio.duration - this.audio.currentTime < 1) {
+        this.next();
```
Comment (jimporter) [inline apps/music/js/ui/views/player_view.js]: I think this should be `this.next(true);`

If you have repeat-song turned on and seek to the end of the song and hit play, I'd expect it to restart the current song, not start the next one.

File: apps/music/js/ui/views/player_view.js (hunk 2 above)
Comment (davidflanagan) [inline apps/music/js/ui/views/player_view.js]: If we have a really short audio file (like a 2 second notification sound) that is only a couple of seconds long, then not allowing pause/play in the last second would be a real problem. I picked 20 pretty much at random. The bug I"m trying to fix here is only bad for song-length m4a files. So with a 20s sound file, things should be okay without this patch, I think.

File: apps/music/js/ui/views/player_view.js (hunk 3 above)
Comment (davidflanagan) [inline apps/music/js/ui/views/player_view.js]: Thanks. I'm trying to replace the 'ended' event here, so and ended uses this.next(true), so clearly that's what I need here, too.

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=48becdd40198b4be079ed59707d2b3144b3a090d

File: apps/music/js/ui/views/player_view.js (hunk 2 above)
Comment (jimporter) [inline apps/music/js/ui/views/player_view.js]: Are you sure this only happens for long files? Maybe we could just do this for the last tenth of a second or something...

**Comment by lightsofapollo** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=82cbbf91e75074e595c4866980be93d2f2d935f3

**Comment by davidflanagan** [comment]
Don't need to run the tests again. The latest push only changed a comment.

**Pull request merged**